tests/test_wasting.py

```diff
@@ -0,0 +1,1062 @@
+                                                                       in_current_episode_before_death].index)
+
+            assert set() == set_of_person_id_in_current_episode_before_recovery.intersection(
+                set_of_person_id_in_current_episode_before_death)
+
+            # WHZ standard deviation of -3, MUAC <115mm, and oedema should cause severe acute malnutrition
+            whz_index = df.index[df['un_WHZ_category'] == 'WHZ<-3']
+            muac_index = df.index[df['un_am_MUAC_category'] == '<115mm']
+            oedema_index = df.index[df['un_am_nutritional_oedema']]
+            assert df.loc[whz_index, 'un_clinical_acute_malnutrition'].eq("SAM").all()
+            assert df.loc[muac_index, 'un_clinical_acute_malnutrition'].eq("SAM").all()
+            assert df.loc[oedema_index, 'un_clinical_acute_malnutrition'].eq("SAM").all()
+
+            # all SAM individuals should have symptoms of wasting
+            assert set(under5_sam).issubset(has_symptoms)
+
+            # All MAM individuals should have no symptoms of wasting
+            assert set(df.index[df.is_alive & (df.age_exact_years < 5) &
+                                (df.un_clinical_acute_malnutrition == 'MAM')]) not in has_symptoms
+
+    popsize = 10_000
+    sim = get_sim(tmpdir)
+    sim.register(DummyModule())
+    sim.make_initial_population(n=popsize)
+    sim.simulate(end_date=end_date)
+
+
+def test_wasting_incidence(tmpdir):
+    """ Check incidence of wasting is happening as expected. """
+    # get simulation object:
+    dur = pd.DateOffset(days=0)
+    popsize = 1000
+    sim = get_sim(tmpdir)
+    # get wasting module
+    wmodule = sim.modules['Wasting']
+
+    sim.make_initial_population(n=popsize)
+    sim.simulate(end_date=start_date + dur)
+
+    # Reset properties of all individuals so that they are well-nourished
+    df = sim.population.props
+    df.loc[df.is_alive, 'un_WHZ_category'] = 'WHZ>=-2'  # not wasted
+    df.loc[df.is_alive, 'un_am_MUAC_category'] = '>=125mm'
+    df.loc[df.is_alive, 'un_am_nutritional_oedema'] = False
+    df.loc[df.is_alive, 'un_clinical_acute_malnutrition'] = 'well'
+    df.loc[df.is_alive, 'un_ever_wasted'] = False
+    df.loc[df.is_alive, 'un_last_wasting_date_of_onset'] = pd.NaT
+
+    # Set incidence of wasting at 100%
+    wmodule.wasting_models.wasting_incidence_lm = LinearModel.multiplicative()
+
+    # Run polling event: check that all children should now have moderate wasting
+    polling = Wasting_IncidencePoll(wmodule)
+    polling.apply(sim.population)
+
+    # Check properties of individuals: should now be moderately wasted
+    under5s = df.loc[df.is_alive & (df['age_years'] < 5)]
+    assert all(under5s['un_ever_wasted'])
+    assert (under5s['un_WHZ_category'].eq('-3<=WHZ<-2')).all()
+    assert (under5s['un_last_wasting_date_of_onset'].eq(sim.date)).all()
+
+
+def test_report_daly_weights(tmpdir):
```

Comment:
very good test. 

don't make this change now, but - for next time - not that this would have been even more readable as a for loop across those four conditions.